src/popover/popover.coffee

```diff
@@ -0,0 +1,108 @@
+###*
+Popover directive displays transcluded content on custom action
+executed on the parent element.
+
+@module shift.components.popover
+
+@param {string} shiftPopover the text of the popover
+@param {string} shiftPopoverTitle (optional) The title of the popover
+@param {string} shiftPopoverTrigger What triggers the popover ? click, hover
+or focus.
+@param {string} shiftPopoverPosition Default to 'bottom', can also be set to
+left, right and top.
+@param {string} shiftPopoverTemplateUrl the template URL for rendering. When
+provided, the text and title attribute are ignored.
+@param {attribute} fixed Use fixed positioning for the the tooltip. To be set
+when the trigger object is also fixed positioned.
+@param {string} shiftPopoverAttachTo a CSS selector where to put the popover
+element. Useful when the popover is appearing in a scrollable area.
+
+@example
+```jade
+input(
+  type = "text"
+  ng-model = "foo"
+  shift-popover = "lorem ipsum"
+  shift-popover-title = "date"
+  shift-popover-trigger = "click|hover|focus"
+  shift-popover-position = "top|bottom|left|right"
+  shift-popover-template-url = "xyz.html"
+  shift-popover-attach-to = ".scrollable.classname"
+  fixed
+)
+```
+###
+
+angular.module 'shift.components.popover', []
+  .directive 'shiftPopover',
+    (
+      $http
+      $compile
+      $templateCache
+    ) ->
+      default_template = '''
+        <div class="popover-container">
+          <h3
+            ng-if = "foo.title"
+            class = "popover-title"
+          > {{ foo.title }} </h3>
+          <p>{{ foo.text }}</p>
+        </div>
+      '''
+
+      ###
+      # Directive definition object
+      ###
+
+      restrict: 'A'
+      scope:
```

Comment:
One possible issue I think there may be with creating an isolate scope for `shift-popover` is when a template is provided that contains bindings to an outer (often a controller) scope. Those bindings will break since the template is compiled with the directive's isolate scope.

By using `scope: true` (a new scope which inherits from a parent) instead and grabbing attribute values off the `attrs` object we can overcome this.